compiler/rustc_session/src/config.rs

```diff
@@ -1679,18 +1679,18 @@ pub fn rustc_optgroups() -> Vec<RustcOptGroup> {
             "",
             "cfg",
             "Configure the compilation environment.\n\
```

Comment:
Suggestion: can you please leave the style guidelines as a doc comment on `rustc_optgroups`? I.e.

```rs
/// Returns ...
///
/// # Option style guidelines
///
/// - `<param>`: Indicates a required parameter
/// - `[param]`: Indicates an optional parameter
/// - `|`: Indicates a mutually exclusive option
/// - `*`: a list element with description
```